File: Rest-Api-V3/deposits-and-withdraw-private/deposit-addresses/deposit-addresses.js
require('dotenv').config();
const axios = require('axios');
const CryptoJS = require("crypto-js");

async function getDepositAddress(asset, network) {
  const apiKey = process.env.API_KEY;
  const secretKey = process.env.API_SECRET;
  const timestamp = new Date().toISOString();
  const nonce = Math.random().toString(36).substring(2);
  const method = 'GET';
  const path = 'api.ox.fun';
  const endpoint = '/v3/deposit-addresses';
  const query = `asset=${asset}&network=${network}`;
  const msgString = `${timestamp}\n${nonce}\n${method}\n${path}\n${endpoint}\n${query}`;

  const sign = CryptoJS.enc.Base64.stringify(CryptoJS.HmacSHA256(msgString, secretKey));

  try {
    const response = await axios.get(`https://${path}${endpoint}?${query}`, {
      headers: {
        'Content-Type': 'application/json',
        'AccessKey': apiKey,
        'Timestamp': timestamp,
        'Signature': sign,
        'Nonce': nonce
      }
    });
    //console.log(response.data);
    if (response.data.success) {
      return response.data.data.address;
    } else {
      console.error('Failed to fetch deposit address');
    }
  } catch (error) {
    console.error('Error making API request:', error.response ? error.response.data : error.message);
  }
}

async function main() {
  const address = await getDepositAddress('OX', 'Solana');
  console.log(address);
}

main();

File: Rest-Api-V3/deposits-and-withdraw-private/deposit-addresses/deposit-addresses.py
import os
import requests
import hmac
import hashlib
import base64
from dotenv import load_dotenv
from datetime import datetime
import random
import string

load_dotenv()

def get_deposit_address(asset, network):
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    timestamp = datetime.utcnow().isoformat()
    nonce = ''.join(random.choices(string.ascii_lowercase + string.digits, k=16))
    method = 'GET'
    path = 'api.ox.fun'
    endpoint = '/v3/deposit-addresses'
    query = f'asset={asset}&network={network}'
    msg_string = f'{timestamp}\n{nonce}\n{method}\n{path}\n{endpoint}\n{query}'

    sign = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')

    try:
        response = requests.get(f'https://{path}{endpoint}?{query}', headers={
            'Content-Type': 'application/json',
            'AccessKey': api_key,
            'Timestamp': timestamp,
            'Signature': sign,
            'Nonce': nonce
        })
        response_data = response.json()
        if response_data.get('success'):
            return response_data['data']['address']
        else:
            print('Failed to fetch deposit address')
    except requests.exceptions.RequestException as error:
        print('Error making API request:', error)

# Example usage

address = get_deposit_address('OX', 'Solana')
print(address)


